caikit_tgis_backend/tgis_connection.py

```diff
@@ -221,13 +224,29 @@ def load_prompt_artifacts(self, prompt_id: str, *artifact_paths: List[str]):
             str,
             artifact_paths=artifact_paths,
         )
-        target_dir = os.path.join(self.prompt_dir, prompt_id)
+
+        target_dir = Path(self.prompt_dir) / prompt_id
         os.makedirs(target_dir, exist_ok=True)
-        for artifact_path in artifact_paths:
+
+        # Don't copy files which are already in the target_dir
+        existing_artifact_names = {f.name for f in target_dir.iterdir()}
```

Comment:
I'm curious, do you happen to know if `iterdir` lazily evaluates the contents of the dir, or if it takes a proactive `os.listdir` and then iterates that (i.e. if the list changes during iteration, what happens)? Not particularly important in this instance, just curious.